vhdbuilder/packer/init-variables.sh

```diff
@@ -301,21 +314,17 @@ if [ "$OS_TYPE" == "Windows" ]; then
 		echo "Generating sas token to copy Windows base image"
 		expiry_date=$(date -u -d "20 minutes" '+%Y-%m-%dT%H:%MZ')
 		sas_token=$(az storage account generate-sas --account-name ${STORAGE_ACCOUNT_NAME} --permissions cw --account-key "$key" --resource-types o --services b --expiry ${expiry_date} | tr -d '"')
-
 		echo "Copy Windows base image to ${WINDOWS_IMAGE_URL}"
 		azcopy-preview copy "${WINDOWS_BASE_IMAGE_URL}" "${WINDOWS_IMAGE_URL}?${sas_token}"
-
 		# https://www.packer.io/plugins/builders/azure/arm#image_url
 		# WINDOWS_IMAGE_URL to a custom VHD to use for your base image. If this value is set, image_publisher, image_offer, image_sku, or image_version should not be set.
 		WINDOWS_IMAGE_PUBLISHER=""
 		WINDOWS_IMAGE_OFFER=""
 		WINDOWS_IMAGE_SKU=""
 		WINDOWS_IMAGE_VERSION=""
-	fi
 
-	# Need to use a sig image to create the build VM
-	if [[ "$MODE" == "sigMode" || "$MODE" == "gen2Mode" ]]; then
-		if [ -n "${WINDOWS_BASE_IMAGE_URL}" ]; then
```

Comment:
Does removing "WINDOWS_BASE_IMAGE_URL" mean that we do not support using raw vhd?

Reply:
We still support. Just moving the conditions `if [ -n "${WINDOWS_BASE_IMAGE_URL}" ]; then` outside and `if [[ "$MODE" == "sigMode" ]]; then` inside raw vhd check. So we will check if raw VHD is used first (**line 309**), and then check if we are using sigMode (**line 327**) (Removed `"$MODE" == "gen2Mode"` since we are reusing sigMode for Gen 2. ) The logic works for both Gen 1 & Gen 2. 

**Context in the master branch:** Line 296 and 318 of [init-variables from the official repo](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/init-variables.sh) both check if we are using raw VHD. If under `sigMode` or `gen2Mode` and no raw VHD is given, line 360 to 365 will be stepped in to initialize the variables. (_empty strings_)

**In this commit**, line 301 through 307 initializes the sigmode source variables. If under `sigMode` and raw VHD, line 386 to 390 will change the sigmode sources accordingly.